org.metaborg.spt.core/src/main/java/org/metaborg/spt/core/SPTUtil.java

```diff
@@ -226,6 +227,31 @@ public static boolean checkATermMatch(IStrategoTerm ast, IStrategoTerm match, IT
                 }
                 result = !stop;
                 break;
+            case TUPLE_CONS:
+                // Tuple([KidMatch, ...])
+                // we ignore any annotations on the AST
+                if(!TermUtils.isTuple(ast)) {
+                    logger.debug("The term is not a tuple.");
+                    result = false;
+                    break;
+                }
+                matchList = (IStrategoList) match.getSubterm(1);
+                if(ast.getSubtermCount() != matchList.size()) {
+                    logger.debug("The number of children {}, did not match the expected number {}",
+                        ast.getSubtermCount(), matchList.size());
+                    result = false;
+                    break;
+                }
+                matchIt = matchList.iterator();
+                stop = false;
```

Comment:
I know it's used in the other cases as well, but I think stop is not a very good name.
Maybe `allSubtermsMatch` would be better